test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java

```diff
@@ -1908,8 +2070,212 @@ public void removingExternalIdOnlyWorksForResearcher() {
         participantService.updateParticipant(STUDY, participant);
         
         assertEquals(EXTERNAL_ID, account.getExternalId());
-        verify(externalIdService, never()).unassignExternalId(any(), any(), any());
-        verify(externalIdService, never()).assignExternalId(any(), any(), any());
+        verify(externalIdService, never()).unassignExternalId(any(), any());
+        verify(externalIdService, never()).commitAssignExternalId(any());
+    }
+    
+    private StudyParticipant.Builder withParticipant() {
+        return new StudyParticipant.Builder().copyOf(PARTICIPANT);
+    }
+    
+    @Test
+    public void createParticipantValidatesUnmanagedExternalId() {
+        BridgeUtils.setRequestContext(new RequestContext.Builder().build());
+        STUDY.setExternalIdValidationEnabled(false);
+        
+        StudyParticipant participant = withParticipant().withExternalId("  ").build();
+        
+        try {
+            participantService.createParticipant(STUDY, participant, false);
+            fail("Should have thrown exception");
+        } catch(InvalidEntityException e) {
+            assertEquals("externalId cannot be blank", e.getErrors().get("externalId").get(0));
+        }
+    }
+    
+    @Test
+    public void createParticipantValidatesManagedExternalId() {
+        BridgeUtils.setRequestContext(new RequestContext.Builder().build());
+        STUDY.setExternalIdValidationEnabled(true);
+        
+        when(accountDao.constructAccount(STUDY, EMAIL, PHONE, EXTERNAL_ID, PASSWORD)).thenReturn(account);
+        try {
+            participantService.createParticipant(STUDY, PARTICIPANT, false);
+            fail("Should have thrown exception");
+        } catch(InvalidEntityException e) {
+            assertEquals("externalId is not a valid external ID", e.getErrors().get("externalId").get(0));
+        }
+    }
+    @Test
+    public void createParticipantNoExternalIdAddedDoesNothing() {
+        BridgeUtils.setRequestContext(new RequestContext.Builder().build());
+        when(accountDao.constructAccount(STUDY, EMAIL, PHONE, null, PASSWORD)).thenReturn(account);
+        StudyParticipant participant = withParticipant().withExternalId(null).build();
+        
+        participantService.createParticipant(STUDY, participant, false);
+        
+        verify(accountDao).constructAccount(STUDY, EMAIL, PHONE, null, PASSWORD);
+        verify(accountDao).createAccount(eq(STUDY), eq(account), any());
```

Comment:
Does this call assignExternalId()? Should we verify it?

Here and in the following:

* createParticipantExternalIdAddedUpdatesExternalId()
* updateParticipantNoExternalIdsOneAddedUpdates()
* updateParticipantAsResearcherNoExternalIdsOneAddedUpdates()
* updateParticipantAsResearcherExternalIdsExistNoneMatchOneAddedUpdates()
* researcherCanChangeExternalIdOnUpdate()

Reply:
Added, and split the researcherCanChangeExternalIdOnUpdate() test for managed and unmanaged external IDs

Reply:
Added and split researcherCanChangeExternalIdOnUpdate() to test managed and unmanaged (unmanaged the external ID field is set, managed as you'd expect, it also persists the external ID record).